pulsar/producer_impl.go

```diff
@@ -90,7 +93,7 @@ func newProducer(client *client, options *ProducerOptions) (*producer, error) {
 	if options.BatchingMaxPublishDelay <= 0 {
 		options.BatchingMaxPublishDelay = defaultBatchingMaxPublishDelay
 	}
-	if options.PartitionsAutoDiscoveryInterval <= 0 {
+	if options.PartitionsAutoDiscoveryInterval <= minPartitionAutoDiscoveryInterval {
```

Comment:
Wouldn't this be a problem if someone has set the value to <= 1 second? There's no error message and the default 1 minute gets used. I would expect at least a warning message. The case of 0 is different since in that case, the default value is expected.

Reply:
Returns an error if PartitionsAutoDiscoveryInterval is less than 1 second